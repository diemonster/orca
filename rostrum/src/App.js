import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from './store/configureStore';

import KubeNav from './components/KubeNav';
import Namespace from './components/namespaces/Namespace';
import Pod from './components/pods/Pod';

const initialState = {
  namespace: {
    namespaceObjects: [],
    namespaceCreateInput: '',
    namespaceDeleteInput: '',
  },
  rolebinding: {
    rolebindingObjects: {},
  },
};

const store = configureStore(initialState);

function App() {
  return (
    <Provider store={store}>
      <Router>
        <div className="app-container">
          <h1>
            Orca: The Kubernetes Permissions Manager
          </h1>
          <div className='kube-container'>
            <KubeNav />
            <Switch>
              <Route exact path='/namespace/' component={Namespace} />
              <Route exact path='/pod/' component={Pod} />
            </Switch>
          </div>
        </div>
      </Router>
    </Provider>
  );
}

export default App;
