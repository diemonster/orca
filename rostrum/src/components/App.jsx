import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { handleAuthentication } from '../actions/authActions';
import NavigationBar from './NavigationBar';
import Namespaces from './Namespaces';
import Rolebindings from './Rolebindings';


export class App extends React.Component {
  componentWillMount() {
    this.handleAuthentication();
  }

  // If the URL has been populated with information from logging in through
  // Auth0, then call the authenticator's handleAuthentication method to
  // set session information
  handleAuthentication() {
    const { dispatchHandleAuthentication, location } = this.props;
    if (/access_token|id_token|error/.test(location.hash)) {
      dispatchHandleAuthentication(location.hash);
    }
  }

  render() {
    const { isAuthenticated } = this.props;

    if (isAuthenticated) {
      return (
        <div className="app-container">
          <NavigationBar />
          <h1>Orca: the Kubernetes permissions manager</h1>
          <div className="kube-container">
            <Namespaces />
            <Rolebindings />
          </div>
        </div>
      );
    }

    return (
      <div className="app-container">
        <NavigationBar />
        <h1>Orca: the Kubernetes permissions manager</h1>
      </div>
    );
  }
}

App.propTypes = {
  dispatchHandleAuthentication: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  location: PropTypes.shape({
    hash: PropTypes.string.isRequired,
  }).isRequired,
};

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
});

const mapDispatchToProps = dispatch => ({
  dispatchHandleAuthentication: urlHash => dispatch(handleAuthentication(urlHash)),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
