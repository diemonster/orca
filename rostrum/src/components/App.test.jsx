import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { App } from './App';

Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  let props;

  beforeEach(() => {
    props = {
      dispatchHandleAuthentication: jest.fn(),
      isAuthenticated: false,
      location: { hash: '' },
    };
  });

  describe('App', () => {
    it('should not render namespaces nor rolebindings when not authenticated', () => {
      const app = shallow(<App {...props} />);
      expect(app).toMatchSnapshot();
    });

    it('should render namespaces and rolebindings when authenticated', () => {
      props.isAuthenticated = true;

      const app = shallow(<App {...props} />);
      expect(app).toMatchSnapshot();
    });

    describe('methods', () => {
      it('handleAuthentication should dispatch an action when the url contains an auth hash', () => {
        props.location = { hash: 'access_token' };
        shallow(<App {...props} />);
        expect(props.dispatchHandleAuthentication).toHaveBeenCalledWith(props.location.hash);
      });

      it('handleAuthentication should not dispatch an action when the url does not contain an auth hash', () => {
        shallow(<App {...props} />);
        expect(props.dispatchHandleAuthentication).not.toHaveBeenCalled();
      });
    });
  });
});
