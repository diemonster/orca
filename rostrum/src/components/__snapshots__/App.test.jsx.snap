// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components App should not render namespaces nor rolebindings when not authenticated 1`] = `
<div
  className="app-container"
>
  <Connect(NavigationBar) />
  <h1>
    Orca: the Kubernetes permissions manager
  </h1>
</div>
`;

exports[`components App should render namespaces and rolebindings when authenticated 1`] = `
<div
  className="app-container"
>
  <Connect(NavigationBar) />
  <h1>
    Orca: the Kubernetes permissions manager
  </h1>
  <div
    className="kube-container"
  >
    <Namespaces />
    <Connect(Rolebindings) />
  </div>
</div>
`;
